testng-eclipse-plugin/src/main/org/testng/eclipse/ui/util/ConfigurationHelper.java

```diff
@@ -165,9 +187,96 @@ public static String getJvmArgs(ILaunchConfiguration configuration) {
   		}
     }
 
-    return result;
-	}
-  
+    jvmArgs.append(" ").append(result);
+
+    addDebugProperties(jvmArgs, configuration);
+
+    switch (ConfigurationHelper.getProtocol(configuration)) {
+    case STRING:
+      jvmArgs.append(" -Dtestng.eclipse.stringprotocol");
+      break;
+    default:
+      break;
+    }
+
+    return jvmArgs.toString();
+  }
+
+  /**
+   * Get the JVM args from maven pom.xmll.
+   * <ul>
+   * Here is the return value of different cases:
+   * <li>no pom.xml -- return empty String</li>
+   * <li>pom.xml exists, but no maven-surefire-plugin or maven-safefail-plugin, in essential, no "argLine" element in the pom.xml -- return empty String</li>
+   * <li>there is one "argLine" element -- return the text content of the "argLine" element</li>
+   * <li>there are more then one "argLine" elements -- return the first "argLine"<profile></li>
+   * </ul>
+   * @param conf
+   * @return
+   * @throws Exception
+   */
+  private static String getVMArgsFromPom(ILaunchConfiguration conf) throws Exception {

```

Comment:
use XPath to get the argLine rather than use Maven eclipse plugin so that we don't hardwire with the m2e plugin.